apps/prairielearn/src/pages/shared/syncHelpers.js

```diff
@@ -53,48 +56,57 @@ export async function gitStatus(locals) {
   return serverJob.jobSequenceId;
 }
 
-function locateImage(image, callback) {
-  docker.listImages(function (err, list = []) {
-    if (ERR(err, callback)) return;
-    for (let i = 0, len = list.length; i < len; i++) {
-      if (list[i].RepoTags && list[i].RepoTags?.indexOf(image) !== -1) {
-        return callback(null, docker.getImage(list[i].Id));
-      }
-    }
-    return callback(new Error(`Unable to find image=${image}`));
-  });
-}
-
-function confirmOrCreateECRRepo(repo, job, callback) {
+/**
+ * @param {string} repo
+ * @param {import('../../lib/server-jobs').ServerJob} job
+ */
+async function ensureECRRepo(repo, job) {
   const ecr = new ECR(makeAwsClientConfig());
   job.info(`Describing repositories with name: ${repo}`);
-  ecr.describeRepositories({ repositoryNames: [repo] }, (err, data) => {
-    let repositoryFound = false;
-    if (err) {
-      job.info(`Error returned from describeRepositories(): ${err}`);
-      job.info('Treating this error as meaning the desired repository does not exist');
+
+  let repositoryFound = false;
+  try {
+    const data = await ecr.describeRepositories({ repositoryNames: [repo] });
+    repositoryFound = !!data.repositories?.find((r) => r.repositoryName === repo);
+  } catch (err) {
+    if (err instanceof RepositoryNotFoundException) {
+      // Repository not found; this is expected.
     } else {
-      repositoryFound = !!_.find(data.repositories, ['repositoryName', repo]);
+      // Something else went wrong; allow it to bubble up.
+      Sentry.captureException(err, {
+        tags: {
+          repository: repo,
+        },
+      });
```

Comment:
Nit (feel free to ignore, prettier can't always find these):
```suggestion
      Sentry.captureException(err, { tags: { repository: repo }});
```
same comment in line 96.

Reply:
Done in 84e2077.